src/reducers/ast.js

```diff
@@ -111,6 +119,10 @@ function update(
       return initialState();
     }
 
+    case "SET_SOURCE_METADATA": {
+      return state.set("metaData", action.metaData);
```

Comment:
ahh, I'd change this to look like empty lines: `return state.setIn(["emptyLines", source.id], emptyLines);`